sdk/core/azure-core-experimental/azure/core/experimental/transport/_httpx_async.py

```diff
@@ -0,0 +1,119 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from collections.abc import AsyncIterator
+from typing import ContextManager, Optional, Any
+
+import httpx
+from azure.core.pipeline import Pipeline
+
+from azure.core.exceptions import ServiceRequestError, ServiceResponseError
+from azure.core.pipeline.transport import AsyncHttpTransport, HttpRequest
+from azure.core.rest._http_response_impl_async import AsyncHttpResponseImpl
+
+from ._httpx import HttpXTransportResponse
+
+
+class AsyncHttpXTransportResponse(HttpXTransportResponse, AsyncHttpResponseImpl):
```

Comment:
Maybe we can put both sync and async transport in one file?

Reply:
We have them split between async and sync in core as well, so I followed that. Any particular reason to merge ?

Reply:
We split it in core because we needed to support both py2 and py3 at that point. :)

I am kind of thinking keeping all httpx stuff in one file may be easier to read and maintain?

Just my personal opinion, it is not a big deal.